core/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -294,9 +295,16 @@ public RestStatus status() {
 
     @Override
     public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
-        builder.startObject(SNAPSHOT);
-        builder.field(NAME, name);
+        // write snapshot info to repository snapshot blob format
+        if (MetaData.CONTEXT_MODE_SNAPSHOT.equals(params.param(MetaData.CONTEXT_MODE_PARAM))) {
```

Comment:
I wouldn't reuse that MetaData parameters here but use new parameters specific for this class.